solr/core/src/java/org/apache/solr/search/CacheOverridesManager.java

```diff
@@ -55,12 +66,12 @@
  * </pre>
  *
  * Entries will be applied as overrides according to the order declared in the array. Hence, later
- * entry overrides earlier for value overlaps
+ * entry overrides earlier for overlaps
  */
 @SuppressWarnings("unchecked")
 public class CacheOverridesManager {
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-  private final ConcurrentMap<String, List<CacheOverrides>> overridesByCacheName =
+  private volatile ConcurrentMap<String, List<CacheOverrides>> overridesByCacheName =
```

Comment:
I don't think this needs to be a concurrent map, since once it becomes a field it's basically immutable. I'd be tempted to do build the new map in a `HashMap`, and then do:
```java
overridesByCacheName = Map.copyOf(newOverridesByCacheName);
```

Reply:
Good call on making it immutable!

Changes made https://github.com/cowpaths/fullstory-solr/pull/265/commits/cbd8440599bb7204a98f525e9af700a8a830c8ed

WDYT? :)

Reply:
Looks good, though as long as we're doing `Map.copyOf(...)`, maybe prefer `Map.of()` (empty map) in place of `Collections.EMPTY_MAP`?